### ISSUE TITLE
Save the Hog into XML

### ISSUE THREAD

**Issue by chenxiemin** [JunshengFu/vehicle-detection#2]
Hi, is it able to save the clf_pickle_all_v1.p into *.xml so that I can use it in c++ detectMultiScale() function?

Sorry for I use this way to ask you the question, thanks very much.

**Comment by JunshengFu** [comment]
Hi @chenxiemin 

If you want to retrain the classifier for svm and save them into xml, you can do it with my code in file `svm_pipeline.py` at line 447 to 552, and just choose the way you want to save the result. You can get the original image with the links I provided in the section named "1.1 Extract Histogram of Oriented Gradients (HOG) from training images".

If you want to reuse my trained svm classifier, you first read it by file `svm_pipeline.py` line 464-467, and then rewrite to xml accordingly. But, I use `LinearSVC` from `sklearn.svm`, if you want to use my model directly, you need to call scikit-learn functions from C++ (e.g. https://stackoverflow.com/questions/30232743/calling-scikit-learn-functions-from-c)

Hope it helps!
.

**Comment by KuppamCR7** [comment]
When I use a different video I get an error in np.polyfit that the input is empty and when I see the values of leftx and lefty in lane.py, they are empty arrays. Is there any fix for this.

**Comment by chenxiemin** [comment]
Hi @JunshengFu 

Would you mind to point me where's the download link of the kitti car database?
I just download several packages from the link: http://www.cvlibs.net/datasets/kitti/eval_object.php

But none of them is the right one. Thanks very much.

**Comment by JunshengFu** [comment]
Hi @chenxiemin 

* [Here](https://s3.amazonaws.com/udacity-sdc/Vehicle_Tracking/vehicles.zip) are example images come from a combination of the GTI vehicle image database, the KITTI vision benchmark suite.
* [Here](https://github.com/udacity/self-driving-car/tree/master/annotations) you can find the Udacity data. In each of the folders containing images there's a csv file containing all the labels and bounding boxes. To add vehicle images to your training data, you'll need to use the csv files to extract the bounding box regions and scale them to the same size as the rest of the training images.

**Comment by JunshengFu** [comment]
Hi @chenxiemin 

* [Here](https://s3.amazonaws.com/udacity-sdc/Vehicle_Tracking/non-vehicles.zip) are some cropped non-vehicle images, and if you need more you can simply crop any images which don't contain vehicles.

**Comment by chenxiemin** [comment]
Hi @JunshengFu 

It's totally enough, thanks for your sharing.
